src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -64,10 +69,68 @@ void parallel_reduce (const TeamMember& team,
 
    result = local_tmp;
   } else {
+    const ValueType initial(result);
     Kokkos::parallel_reduce(Kokkos::TeamThreadRange(team, begin, end), lambda, result);
+    result += initial;
   }
 }
 
+/*
+ * Computes a reduction over the view described by 'input' for entries 'input([begin,end))'
+ * The variable 'input' should be an operator s.t. 'input(k)' returns a Pack of size 'vector_size'.
+ * The variable 'total_size' represents the total number of 'ValueType' entries in the view. The
+ * computed reduction is added to 'result'.
+ */
+template <bool Serialize, typename TeamMember, typename InputProvider, typename ValueType>
+static KOKKOS_INLINE_FUNCTION
+void view_reduction (const TeamMember& team,
+                     const int total_size,
+                     const int vector_size,
+                     const int begin,
+                     const int end,
+                     const InputProvider& input,
+                     ValueType& result)
+{
+  const int  n_packs        = (total_size + vector_size - 1)/vector_size;
+  const bool has_garbage    = (end==n_packs && total_size % vector_size != 0 ? true : false);
+  const int  loop_end       = (has_garbage ? end-1 : end);
+  const int  last_pack_size = (has_garbage ? total_size % vector_size : vector_size);
+
+  // For parallel computation, perform a packed reduction
+  if (!Serialize) {
+    impl::parallel_reduce<Serialize>(team, begin, loop_end,
```

Comment:
I don't think this can be right. `begin` is the scalar begin level, just like `end` is the scalar end level. Pack arithmetic must be done on it just like for `end`.

Reply:
For clarity: `begin` and `end` should be _scalar_ indices, not pack indices. The physics param writer thinks in terms of model levels, not packs.